### PR TITLE
Model migration integration tests

### PR DESCRIPTION
**Pull request by SimonRichardson**
## Description of change

Adding model migration integration tests, before adding CMR model
migrations. This should help re-affirm what's required from a following
set of commands.

## QA steps

```sh
cd tests && ./main.sh model test_model_migration
```

### CODE REVIEW

**Comment by hmlanigan** [comment]
Should I be able to do this:
$ ( cd tests && ./main.sh model )

==> Checking for dependencies
==> TEST BEGIN: test_model
./main.sh: 238: ./main.sh: test_model: not found
==> Cleaning up

==> TESTS DONE: test_model
==> Test result: failure
==> TEST COMPLETE

**Comment by SimonRichardson** [comment]
Nope, you need the #10862 before that will work

File: tests/suites/model/charms/ubuntu/metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+summary: A pristine Ubuntu Server
+maintainer: Charmers <juju@lists.ubuntu.com>
+description: |
+  This simply deploys Ubuntu Server.
+categories:
+     - misc
+series:
+  - trusty
+  - xenial
+  - artful
```
Comment (hmlanigan) [inline tests/suites/model/charms/ubuntu/metadata.yaml]: suggest removing artful and adding eoan and focal.

File: tests/suites/model/charms/ubuntu/metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+name: ubuntu
```
Comment (hmlanigan) [inline tests/suites/model/charms/ubuntu/metadata.yaml]: suggest we add a few endpoints, they don't need to be used.

**Comment by hmlanigan** [comment]
$$merge$$

**Comment by hmlanigan** [comment]
oops, sorry, typed merge in the wrong window.  :-(

**Comment by SimonRichardson** [comment]
$$merge$$

**Pull request merged**